ext/standard/basic_functions.stub.php

```diff
@@ -1879,9 +1914,15 @@ function base64_decode(string $string, bool $strict = false): string|false {}
 
 function constant(string $name): mixed {}
 
+/**
+ * @compile-time-eval
+ */
 function ip2long(string $ip): int|false {}
```

Comment:
ip2long & long2ip both can call into inet_pton & inet_ntop.
https://man7.org/linux/man-pages/man3/inet_pton.3.html
https://man7.org/linux/man-pages/man3/inet_ntop.3.html

Both have the locale attribute, indicating they might depend on the locale.

Reply:
How can ip2long depend on locale?

Reply:
I don't know, but the C standard specifies it is possible.

Reply:
Google for "inet_ntop c source". When I look at glibc impl, the output is coded like:
```
fulllen = snprintf(tmp, sizeof(tmp), "%d.%d.%d.%d", ap[0], ap[1], ap[2], ap[3]);
```

Can `%d` with input `0` - `255` integer be formatted differently with some special locale?

Reply:
> Can %d with input 0 - 255 integer be formatted differently with some special locale?

I don't think so. But since the manpage mentions that it can read from the locale, the implementations are technically allowed to do that. So even if the current implementations don't depend on the locale, when the implementation changes this would introduce problems for us.
I could be wrong about the locale attribute of course, but this is at least my understanding.

Reply:
I would say we should be ok here. The output format is quite well defined and tested.

Reply:
It's your job to prove that the functions actually are safe in 100% of cases, not the reviewers job to point out that they are unsafe. Frankly with the number of changes that needed to be reverted from this PR, I feel like it is not reviewable in practice with a sufficient degree of confidence that it is correct. Perhaps it should be split into several smaller PRs, each focusing on a small set of functions and outlining why:

1. The change is safe.
2. The cost/benefit ratio is worth the change.

Reply:
You are right, but also prooving any atom in the universe is not possible. I might be wrong with these functions, but I need also some example explaning the locale dependence. I expect well defined packed/unpacked format.

About other functions - please comment here.

Reply:
Yes. See also: https://man7.org/linux/man-pages/man7/attributes.7.html

Reply:
Yes, but how, what config can cause the result to be different?

Reply:
@nielsdos please follow https://stackoverflow.com/questions/76108650, I asked this on SO forum and it seems `MT-Safe locale` flag on these C functions is about type and thread safety, but the result should not be different.

Reply:
@nielsdos can you please give this PR a grteen light?

Reply:
I was still not provided an environment where the result can be different of these functions, but as the CTE cannot be used to initialize const prop/param defaults, I agree, pursuing it now is not required either. Reverted. Are there any other functions that should not have CTE flag in this PR?

Reply:
I'm out of time for today so I just took a brief look:

* strnatcmp and strnatcasecmp can depend on locale because they use toupper internally. (I believe it was some Turkish locale which had a special conversion from lowercase i to an I with an accent on it?) So those two functions don't look CTE-safe.
* metaphone() uses toupper() internally, so a potential same problem as above.
* soundex() same toupper() problem
* hebrev() is not safe for CTE because it calls ispunct(). ispunct() is locale dependent according to `man 3 ispunct`: "The details of what characters belong to which class depend on the locale."
* Side note: The md5() & sha1() case are safe, not sure if these will come up a lot with constants in real code though

Reply:
All functions removed.